etc/spack/defaults/config.yaml

```diff
@@ -0,0 +1,65 @@
+# -------------------------------------------------------------------------
+# This is the default spack configuration file.
+#
+# Settings here are versioned with Spack and are intended to provide
+# sensible defaults out of the box. Spack maintainers should edit this
+# file to keep it current.
+#
+# Users can override these settings by editing the following files.
+#
+# Per-spack-instance settings (overrides defaults):
+#   $SPACK_ROOT/etc/spack/config.yaml
+#
```

Comment:
`etc/spack/config.yaml` should be added to `.gitignore`.  We don't want to normally check site configuration back into Spack.  (If this is not added, `config.yaml` will end up accidentally in some PRs).